lib/vdsm/storage/operation.py

```diff
@@ -150,7 +151,17 @@ def _finalize(self, out, err):
             `RuntimeError` if operation state is invalid
         """
         rc = self._proc.returncode
-        log.debug(cmdutils.retcode_log_line(rc, err))
+        log_level = logging.DEBUG
+        log_msg = cmdutils.retcode_log_line(rc, err)
+        if self._warn_on_error_output and err:
+            log_level = logging.WARNING
+            if rc == 0:
+                # Log the command if it succeeded. Otherwise, the command
+                # will appear in the raised Error.
+                log_msg = \
+                    f"Command {self._cmd} suceeded with warnings: {log_msg}"
+        log.log(log_level, "%s", log_msg)
```

Comment:
This is little bit too complicated. I agree that we want to keep the old log unless
the command succeeded with non empty stderr. How about:

     if rc == 0 and self._warn_stderr and err:
         log.warning("Command %s succeeded with warnings: %s", self._cmd, err)
     else:
         log.debug(cmdutils.retcode_log_line(rc, err))

Reply:
> Either way, I would take the chance to change it to printf style:

A good change.  If you make another update, please put it in a separate commit.